DDInstall/DDInstall.psm1

```diff
@@ -0,0 +1,192 @@
+function Install-DDAgent
+{
+    param (
+        [Parameter()][ValidateSet('DotNet')][String[]]$WithAPMTracers,
+        [String] $AgentInstallURL,
+        [Parameter(Mandatory)] [String]$ApiKey,
+        [String]$Site,
+        [String]$Tags,
+        [String]$Hostname,
+        [String]$DDAgentUsername,
+        [String]$DDAgentPassword,
+        [String]$ApplicationDataDirectory,
+        [String]$ProjectLocation,
+        [String]$InstallLogPath
+    )
+
+    $currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())
+    if (-not $currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator))
+    {
+        $exception = [Exception]::new("Administrator priviledges required.")
+        $PSCmdlet.ThrowTerminatingError([System.Management.Automation.ErrorRecord]::new($exception, "FatalError", [Management.Automation.ErrorCategory]::InvalidOperation, $null))
+    }
+
+    # Enable TLS 1.2
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
+
+    Write-Host "Downloading Datadog Windows Agent installer"
+    $url = "https://s3.amazonaws.com/ddagent-windows-stable/datadog-agent-7-latest.amd64.msi"
+    if ($PSBoundParameters.ContainsKey('AgentInstallURL'))
+    {
+        $url = $AgentInstallURL
+    }
+    downloadAsset -url $url -outFile "datadog-agent.msi"
+
+    Write-Host "Installing Datadog Windows Agent"
+    $installerParameters = formatAgentInstallerParameters -params $PSBoundParameters
+    $installResult = Start-Process -Wait msiexec -ArgumentList "/qn /i datadog-agent.msi $installerParameters" -PassThru
+    if ($installResult.ExitCode -ne 0)
+    {
+        $logFile = "%TEMP%\MSI*.LOG"
```

Comment:
instead of not knowing what the exact log path if the user didn't set one, we could create our own default using [New-TemporaryFile](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-temporaryfile?view=powershell-7.4), which would let us provide the exact path to the user in the case of an error.

Reply:
another great idea!